objects/SetupChecker/script.lua

```diff
@@ -2471,41 +2434,46 @@ function updateDropdownSelection(id, value)
     end)
 end
 
-function addExpansionToggle(value)
-    return matchRecurse("expansionsRow", function (t)
-        table.insert(t.children[1].children[1].children, {
-            tag="Toggle",
-            value=value,
-            attributes={id = value, onValueChanged = "toggleExpansion"},
-            children={},
-        })
+-- Update a 2-width grid of toggle buttons to the have the given items and states
+-- @param id The id of the row containing the grid of toggles
+-- @param values A table mapping toggle names to boolean values
+-- @param onValueChanged The name of the function to call when a toggle is clicked
+function updateToggleGrid(id, values, onValueChanged)
+    return matchRecurse(id, function (t)
+        t.children[1].children[1].children = {}
+        for name,isOn in pairs(values) do
+            table.insert(t.children[1].children[1].children, {
+                tag="Toggle",
+                value=name,
+                attributes={id = name, onValueChanged = onValueChanged, isOn = tostring(isOn)},
+                children={},
+            })
+        end
         local count = #t.children[1].children[1].children
-        t.attributes["preferredHeight"] = math.floor((count + 1) / 2) * 60 + 0.1
+        t.attributes["preferredHeight"] = math.ceil(count / 2) * 60 + 0.1
     end)
 end
-function addEventToggle(value)
-    return matchRecurse("events", function (t)
-        table.insert(t.children[1].children[1].children, {
-            tag="Toggle",
-            value=value.." Events",
-            attributes={id = value.." Events", onValueChanged = "toggleEvents"},
-            children={},
-        })
-        local count = #t.children[1].children[1].children
-        t.attributes["preferredHeight"] = math.floor((count + 1) / 2) * 60 + 0.1
-    end)
+function updateExpansionToggles()
+    local exps = Global.getTable("expansions")
+    local values = {}
+    for name,_ in pairs(expansions) do
+        -- The Global expansions table stores nil for disabled expansions
+        -- We want a boolean false, so we explicitly check for equality to true
+        values[name] = (exps[name] == true)
```

Comment:
The explanation of this (and your following comment) rely on a bit of an oddity in how you store the expansions. They're three relevant tables.
- `SetupChecker` has an `expansions` table, that maps expansion names to bag GUIDs.
- `Global` has an `expansions` table that maps expansion names to `true` if they're enabled or `nil` if they're not.
- `Global` has an `events` table that maps expansion names to `true` if the relevant events are enable or `nil` if they're not.
Further, note that when iterating over a table in Lua, any keys that are mapped to the `nil` value are treated as not being in the table.

When you remove an expansion, it's removed from `SetupChecker`'s `expansions` table. Because this loop iterates over that `expansions` table, a removed expansion doesn't come up in the iteration, and thus isn't added to the UI. The `exps` table (which is `Global`'s `expansions` table) is just used to check whether the expansion is enabled or not, and thus whether the box should be ticked or not.

I have tested it, and if you delete an expansion's bag, the relevant tickbox vanishes from the UI.